Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -1566,4 +1566,76 @@ function Set-ToolsRepo
             Write-Host "Successfully set tools-repo on all hosts for datastore $ds_name"
         }
     }
-}
\ No newline at end of file
+}
+
+<#
+.Synopsis
+    Set vSAN compression and deduplication on a cluster or clusters. If deduplication is enabled then compression is required.
+    The default cluster configuration is deduplication and compression but the customer can change that.
+    Choosing neither compression nor deduplication will disable both.
+    This requires action on every physical disk and will take time to complete.
+
+.EXAMPLE
+    Set-vSANCompressDedupe -Cluster "cluster-1,cluster-2" -Compression $true
+
+    Set-vSANCompressDedupe -Cluster "cluster-1,cluster-2" -Deduplication $true
+
+    Set-vSANCompressDedupe -Cluster "cluster-1,cluster-2"
+
+    Set-vSANCompressDedupe -Cluster "all"
+#>
+
+function Set-vSANCompressDedupe
+{
+    [AVSAttribute(120, UpdatesSDDC = $true)]
+    param(
+        [Parameter(Mandatory = $true)]
+        [String]$Cluster,
+        [Parameter(Mandatory = $false,
+        HelpMessage = "Enable compression and deduplication.")]
+        [bool]$Deduplication = $false,
+        [Parameter(Mandatory = $false,
+        HelpMessage = "Enable compression only.")]
+        [bool]$Compression = $true,
+        [Parameter(Mandatory = $false,
+        HelpMessage = "Neither compression nor deduplication.")]
+        [bool]$Neither = $false
+    )
+
+    # $cluster is an array of cluster names or "all"
+    If ($Cluster -eq "all")
+    {
+        $Clusters = Get-Cluster
+    }
+    else
+    {
+        foreach ($cluster_each in ($Cluster.split(",")).Trim())
+        {
+            $Clusters += Get-Cluster -Name $cluster_each
+        }
+    }
+
+    foreach ($Cluster in $Clusters)
+    {
+        $cluster_name = $Cluster.Name
+
+        If ($Deduplication -eq $true)
+        {
+            # Deduplication requires compression
+            Write-Host "Enabling deduplication and compression on $cluster_name"
+            Set-VsanClusterConfiguration -Configuration $cluster_name -SpaceEfficiencyEnabled $true
+        }
+        elseif ($Compression -eq $true)
+        {
+            # Compression only
+            Write-Host "Enabling compression on $cluster_name"
+            Set-VsanClusterConfiguration -Configuration $cluster_name -SpaceCompressionEnabled $true
+        }
+        elseif ($Neither -eq $true)
+        {
+            # Neither compression nor deduplication
+            Write-Host "Disabling compression and deduplication on $cluster_name"
+            Set-VsanClusterConfiguration -Configuration $cluster_name -SpaceEfficiencyEnabled $false
```

Comment:
This doesn't actually disable space compression?

I don't think this whole thing should be done with elseif, I'd call `Set-VsanClusterConfiguration` once and pass both parameters to it.

Reply:
You can only have one: 
SpaceEfficiencyEnabled $true -- enables compression and deduplication
SpaceCompressionEnabled $true -- enables compression and disables deduplication
SpaceEfficiencyEnabled $false -- disables compression and deduplication

Reply:
I see, my sense is that because Compression and Deduplication are not presented as mutually exclusive options (and we don't really have a way to do that atm) they shouldn't be treated as mutually exclusive here, which is what `elseif` would do, otherwise we'd be dealing with support tickets to explain it.
If I'm reading the docs right, this was very odd choice on vmware's part, but for one, I'd stick with their names, and for the sequence how about this:

```powershell
Set-VsanClusterConfiguration -Configuration $cluster_name -SpaceEfficiencyEnabled ($Compression -or $Efficiency)
Set-VsanClusterConfiguration -Configuration $cluster_name -SpaceCompressionEnabled $Compression
```

Reply:
I tried to [spell it out in the synopsis](https://github.com/Azure/Microsoft.AVS.Management/pull/133/files#diff-fa3c6c03125bdcf1960d19eb4bbf4b87fc503b8eaebbeefb5b836b1730978bb0R1572-R1576) but perhaps did a poor job.

Reply:
...radio buttons would be best.  😀